static/js/stream_data.js

```diff
@@ -478,17 +478,19 @@ exports.receives_notifications = function (stream_id, notification_name) {
 
 exports.update_calculated_fields = function (sub) {
     sub.is_admin = page_params.is_admin;
+    sub.can_perform_admin_actions = page_params.is_admin || sub.is_stream_admin;
     // Admin can change any stream's name & description either stream is public or
```

Comment:
`sub.is_admin` does not feel correct because it may be interpreted as stream admin or realm admin both. 
I have not changed/removed it yet because it is used to check whether the user is allowed to delete or messages in topic or move a topic and wanted to have a discussion on whether we would allow stream admins to perform these tasks. We can change this after we have decided or we can just replace the places where we use `sub.is_admin` by `page_params.is_admin`

Reply:
Yes, agreed.  We might want to head off this sort of confusion on the other end by renaming `sub.is_admin` to `sub.is_realm_admin` everywhere in a preparatory refactoring commit.  (Or maybe we can just remove that field altogether?  I'm not sure it needs to be copied onto stream objects)?

`sub.can_perform_admin_actions` should be `sub.can_administer_stream`, I think.

So maybe the refactoring is just a prep commit that does

s/sub.is_admin/sub.can_administer_stream/, and then this commit just adds `sub.is_stream_admin` and changes how `sub.can_administer_stream` is calculated?

Reply:
I think we can just replace sub.is_admin with page_params.is_admin. There are only few places where we use this.

Reply:
Remember to check `git grep is_admin static/templates/` -- we might pass a `sub` to a template in some places.

But I think my proposal achieves that same goal while having a nice valid `sub.can_administer_stream` that we'll find useful.

Reply:
Looks like `static/templates/subscription_settings.hbs` has it -- so that's why it's there.  In any case I think we're agreed we should just remove `sub.is_admin` in favor of something better.

Reply:
`is_admin` in `static/templates/subscription_settings.hbs` will be replcaed by `can_administer_stream` as we would allow stream admin to deactivate the stream.
I have checked and the only place we use is_admin is in `static/templates/topic_sidebar_actions` and in `stream_popover.js` to render the template mentioned above.